fastparquet/writer.py

```diff
@@ -571,6 +574,12 @@ def write(filename, data, partitions=[0], encoding="PLAIN",
                     fmd.row_groups.append(rg)
             fmd.num_rows = sum(rg.num_rows for rg in fmd.row_groups)
         else:
+            if write_index or write_index is None and not (
+                    isinstance(data.index, pd.RangeIndex) and
+                    data.index._start == 0 and
+                    data.index._stop == len(data.index) and
+                    data.index._step == 1 and data.index.name is None):
+                data = data.reset_index()
```

Comment:
It sets it up decently and errs in an informative way if a conflict arises.

I don't need metadata for immediate use with dask.dataframe, so I plan to leave it for future work.